packages/truffle-core/lib/commands/debug.js

```diff
@@ -303,46 +305,188 @@ var command = {
+
+            if(isNaN(line))
+            {
+              config.logger.log("Line number must be an integer");
+              return;
+            }
+
+            breakpoint.sourceId = currentSourceId;
+            breakpoint.line = line-1; //adjust for zero-indexing!
+          }
+
+          //having constructed the breakpoint, here's now a usable-readable
+          //message describing its location
+          let locationMessage;
+          if(breakpoint.node !== undefined)
+          {
+            locationMessage = `this point in line ${breakpoint.line+1}`;
+            //+1 to adjust for zero-indexing
+          }
+          else if(breakpoint.sourceId !== currentSourceId)
+          {
+            //name the source as the user entered it
+            //note: we should only be in this case if a source was entered!
+            //if no source as entered and we are here, something is wrong
+            locationMessage = `line ${breakpoint.line+1} in ${sourceName}`;
+            //+1 to adjust for zero-indexing
+          }
+          else
+          {
+            locationMessage = `line ${breakpoint.line+1}`
+            //+1 to adjust for zero-indexing
+          }
 
-          config.logger.log("Breakpoint added.");
+          //one last check -- does this breakpoint already exist?
+          let alreadyExists = breakpoints.filter(
+              (existingBreakpoint) =>
+              existingBreakpoint.sourceId === breakpoint.sourceId &&
+              existingBreakpoint.line === breakpoint.line &&
+              existingBreakpoint.node === breakpoint.node //may be undefined
+          ).length > 0;
+
+          //NOTE: in the "set breakpoint" case, the above check is somewhat
+          //redundant, as we're going to check again when we actually make the
+          //call to add or remove the breakpoint!  But we need to check here so
+          //that we can display the appropriate message.  Hopefully we can find
+          //some way to avoid this redundant check in the future.
+
+          //if it already exists and is being set, or doesn't and is being
+          //cleared, report back that we can't do that
+          if(setOrClear === alreadyExists)
+          {
+            if(setOrClear)
+            {
+              config.logger.log(`Breakpoint at ${locationMessage} already exists.`);
+              return;
+            }
+            else
+            {
+              config.logger.log(`No breakpoint at ${locationMessage} to remove.`);
+              return;
+            }
+          }
+         
+          //finally, if we've reached this point, do it!
+          session.setOrClearBreakpoint(breakpoint,setOrClear);
+
+          //finally, report back to the user on what happened
+          if(setOrClear)
+          {
+            config.logger.log(`Breakpoint added at ${locationMessage}.`);
+          }
+          else
+          {
+            config.logger.log(`Breakpoint removed at ${locationMessage}.`);
+          }
```

Comment:
Ah so I just tried this feature out and realized something that might be nice: an extra blank line after these added/removed messages.

This looks a bit cluttered:
```
debug(develop:0x91c817a1...)> b SquareLib:5
Breakpoint added at line 5 in SquareLib.sol.
debug(develop:0x91c817a1...)> b +10
Breakpoint added at line 23.
debug(develop:0x91c817a1...)> B +10
Breakpoint removed at line 23.
```

Compared with:
```
debug(develop:0x91c817a1...)> b SquareLib:5
Breakpoint added at line 5 in SquareLib.sol.

debug(develop:0x91c817a1...)> b +10
Breakpoint added at line 23.

debug(develop:0x91c817a1...)> B +10
Breakpoint removed at line 23.
```